src/main/java/org/symphonyoss/symphony/messageml/elements/Keyword.java

```diff
@@ -53,9 +52,9 @@ public void validate() throws InvalidInputException {
     if (this.tag == null) {
       throw new InvalidInputException("The attribute \"tag\" is required");
     }
-
-    if (!this.tag.matches(KEYWORD_PATTERN)) {
-      throw new InvalidInputException("Keywords may only contain alphanumeric characters, underscore, dot and dash");
+    String pattern = getTagPattern();
+    if (!this.tag.matches(pattern)) {
+      throw new InvalidInputException(this.getClass().getSimpleName()+" must match the following pattern : "+pattern);
```

Comment:
A more accurate error message would be "Values of the attribute \"tag\" for the element " + this.MESSAGEML_TAG + " must match the pattern " + pattern + ".".